pdns/dnsname.cc

```diff
@@ -134,16 +135,19 @@ void DNSName::packetParser(const char* qpos, int len, int offset, bool uncompres
         throw std::range_error("Found compressed label, instructed not to follow");
 
       labellen &= (~0xc0);
-      int newpos = (labellen << 8) + *(const unsigned char*)pos;
+      size_t newpos = (labellen << 8) + *(const unsigned char*)pos;
 
       if(newpos < offset) {
-        if(newpos < (int) minOffset)
+        if(newpos < static_cast<size_t>(minOffset)) {
```

Comment:
I think this cast is not needed, as `newpos` and  and `minOffser` are both unsigned types.

Reply:
Done, thanks.